### ISSUE TITLE
Immediate Save node after drag drop.

### ISSUE THREAD

**Issue by vinayakbelwalkar** [dabeng/OrgChart#256]
Hi,
    I have implemented saving of chart nodes after multiple drag drop by using getHierarchy() method. Now i want event to save the node immediately after drag drop to another parent.
Can you suggest events in the javascript file so that i can handle it? Also can i write my custom events to handle this kind of scenario?

Thanks,
Vinayak belwalkar

**Comment by vinayakbelwalkar** [comment]
Hi.. is anyone having idea to implement this change please?

**Comment by vinayakbelwalkar** [comment]
Hi.. is anyone having idea to implement this change please?

**Comment by mahbubcubd** [comment]
Hi.. can you please provide the code for saving the chart data into json datascource after multiple drag drops by using getHierarchy() method?

**Comment by vinayakbelwalkar** [comment]
Hi,
   I have written function in java script to get content of chart and then passed to json method as string parameter

 function Save() {
        var hierarchy = $('#ControlID').orgchart('getHierarchy');
        var tree = JSON.stringify(hierarchy, null, 2);
        // json call

       }
and at the controller method i have parse it as list of class as-

  JavaScriptSerializer jsonSerializer = new JavaScriptSerializer();
  DragResponse dragResponse = jsonSerializer.Deserialize<DragResponse>(tree);

and then after i have iterated to store in the database. Hope this makes sense.
Let me know your consideration.

Thanks,
Vinayak Belwalkar

**Comment by toshetoo** [comment]
Hello, 

You can listen to the ".drop" event of the node and once the event is triggered, you can perform the "save" operation.

**Comment by vinayakbelwalkar** [comment]
@toshetoo: 
Yes. I have handled this event but i can got id if dragged at node itself ( empty area of node). But there are multiple divisions for displaying photo/name/deptment in the node itself and if i dragged in to any of these division i cannot get id of the node.

I have tested with following events
- dragstart
- dragover
- dragend
- drop
- click

Is there is having any way so that i can get the id of node?

Thanks,
Vinayak Belwalkar

**Comment by toshetoo** [comment]
@vinayakbelwalkar  Can you give a plunker example of your code?

**Comment by whittinghamj** [comment]
I know this is an old thread but I would like some code examples please.

Even a one time save on hitting a 'save' button would be awesome.

Thanks
